apps/arena/lib/arena/game/skill.ex

```diff
@@ -268,14 +262,21 @@ defmodule Arena.Game.Skill do
     }
 
     ## TODO: Magic number needs to be replaced with state.game_config.game.tick_rate_ms
-    speed = Physics.calculate_speed(entity.position, target_position, leap.duration_ms) * 30
+    duration = Physics.calculate_duration(entity.position, target_position, leap.speed) * 30
+
+    Process.send_after(
+      self(),
+      {:stop_leap, entity.id, entity.aditional_info.base_speed, leap.on_arrival_mechanic},
+      duration
+    )
```

Comment:
we're calculating the duration twice, maybe we could include it in the skill params map that `do_mechanic` receives and use that, what do you think? To do so you need to add the duration to the send message of `delayed_skill_mechanics`

Reply:
Nice catch, moved it as you said